src/field_theory/subfield.lean

```diff
@@ -61,21 +63,26 @@ subfield, subfields
 open_locale big_operators
 universes u v w
 
-variables {K : Type u} {L : Type v} {M : Type w} [field K] [field L] [field M]
+variables {K L M : Type*}
 
 set_option old_structure_cmd true
 
-/-- `subfield R` is the type of subfields of `R`. A subfield of `R` is a subset `s` that is a
-  multiplicative submonoid and an additive subgroup. Note in particular that it shares the
-  same 0 and 1 as R. -/
-structure subfield (K : Type u) [field K] extends subring K :=
-(inv_mem' : ∀ x ∈ carrier, x⁻¹ ∈ carrier)
+/-- `subfield K` is the type of subfields of `K`.
+  A subfield of a division ring `K` is a subset `s` that is
+  an additive subgroup and a commutative multiplicative submonoid
+  closed under multiplicative inverses.
+  Note in particular that it shares the same `0` and `1` as `K`. -/
+structure subfield (K : Type u) [division_ring K] extends subring K :=
+(inv_mem'  : ∀ x ∈ carrier, x⁻¹ ∈ carrier)
+(mul_comm' : ∀ x y, x ∈ carrier → y ∈ carrier → x * y = y * x)
```

Comment:
Does anything actually need this property? I'd be inclined to remove it.

Reply:
Fields are commutative while division rings need not be, so this is required. Of course you may want to define `sub_division_ring` and have `subfield` extend that. Is it worth it?

Reply:
Well, for `subgroup` we don't have a separate structure for commutative and non-commutative subgroups; we have the non-commutative structure, and then the predicate `subgroup.normal`. Should we do something similar here?

Reply:
This feels like a "modules are actually semimodules" situation where we can call it a subfield even though it's really a sub-division_ring.

Reply:
This is more analogous to defining the type of commutative subgroups of an arbitrary group. There is no way around the commutativity assumption.

Reply:
There's a zulip thread somewhere about how to handle commutative subgroups, it's probably worth finding.

Reply:
I made an attempt at removing it in #8919, for comparison.

Reply:
I agree that the definition of `subfield` should contain the `mul_comm` hypothesis, otherwise it is just lying. Though it would be interesting to see exactly where we need this hypothesis. I suspect that most of our results would work just as well for `subdivision_ring`.